base/abstractarray.jl

```diff
@@ -208,6 +208,13 @@ reshape(a::AbstractArray, dims::Int...) = reshape(a, dims)
 
 ## from general iterable to any array
 
+function copy!(dest, src)
+    for x in src
+        push!(dest, x)
```

Comment:
Every other `append!` method is equivalent to this, doing `for x in src; push!(dest, x); end`, right?

There are other non-abstractarray collections other than dicts or sets so defining this with such a loose signature seems wrong. Why should `copy!` append to existing keys for Associative collections but overwrite existing values for arrays?